include/envoy/secret/secret.h

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <memory>
+#include <string>
+
+#include "envoy/common/pure.h"
+
+#include "common/protobuf/protobuf.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * An instance of the secret.
+ */
+class Secret {
+public:
+  virtual ~Secret() {}
+
+  /**
+   * @return protobuf message that initialized the secret.
+   */
+  virtual const Protobuf::Message& message() const PURE;
```

Comment:
I don't think this function is actually used anywhere. It can be removed from the interface.

Reply:
Removed